nengo_gui/examples/tutorial/25-spa-parse.py

```diff
@@ -4,53 +4,67 @@
-# by B.  It will remember the two terms, store them, and then respond
-# appropriately by sending B to its hand.  It can also SAY D, where the vector
-# for D will be set to the speech system.
+# by HI.  It will remember the two terms, store them, and then respond
+# appropriately by sending HI to its hand.  It can also SAY BYE, where the
+# vector for BYE will be set to the speech system.
 
 # The parsing process is implemented with the first two actions.  The first
-# action looks for verbs (WRITE or SAY) and sends them into the memory system
-# while binding them with VERB at the same time.  The second action looks for
-# nouns (A or B or C or D or E) and sends them to the memory while binding
-# them with NOUN.  Notice that the actions can looks for multiple different
-# things at the same time by using the + operation.  Overall, this shows how
-# we can use a single input modality (vision) and treat the information
-# in different ways as appropriate.
-#
+# action looks for verbs (WRITE or SAY) and sends them into the memory for
+# verbs. The second action looks for nouns (YES or NO or HI or BYE or OK) and
+# sends them to the memory for nouns.  The noun stored in the noun memory binds
+# with the semantic pointer NOUN and the verb stored in the verb memory binds
+# with the semantic pointer VERB in the phrase state.  This binding is
+# implemented with cortical actions. Notice that the actions can looks for
+# multiple different things at the same time by using the + operation.
+# Overall, this shows how we can use a single input modality (vision) and treat
+# the information in different ways as appropriate.
+
 # The last two actions deal with recognizing and executing actions.  The
 # first one looks for the VERB WRITE, and if it sees this it will take
-# whatever is in memory, extract out the NOUN, and send that to the hand.
+# whatever is in phrase, extract out the NOUN, and send that to the hand.
 # Furthermore, it will only do this if it doesn't currently see anything in
 # vision (via the subtraction).  This is to make sure it waits until it is
 # getting no more visual input before responding.
 
-# To test the model, try presenting WRITE to the vision Buffer.  The memory
-# should fill with WRITE*VERB.  Now change the vision input to B.  The memory
-# should now fill with NOUN*B.  If you get rid of the visual input (by setting
-# the value to nothing), it shuold sent B to the hand.  If you try the same
+# To test the model, try presenting WRITE to the vision State.  The phrase
+# should fill with WRITE*VERB.  Now change the vision input to HI.  The phrase
+# should now fill with NOUN*HI.  If you get rid of the visual input (by setting
+# the value to nothing), it shuold sent HI to the hand.  If you try the same
 # thing with SAY it should send the result to speech.
 
 import nengo
 import nengo.spa as spa
 
-D = 32  # the dimensionality of the vectors
+D = 64  # the dimensionality of the vectors
```

Comment:
If it's possible to get away with 32 dimensions, I'd prefer it, as then it'll work okay on slower computers....